apps/src/blockly/addons/cdoUtils.ts

```diff
@@ -317,13 +317,39 @@ export function bindBrowserEvent(
 export function isWorkspaceReadOnly() {
   return false; // TODO - used for feedback
 }
+/**
+ * Checks if any block type's usage count exceeds its defined limit and returns
+ * the type of the first block found to exceed.
+ * @returns {string | null} The type of the first block that exceeds its limit,
+ * or null if no block exceeds the limit.
+ */
+export function blockLimitExceeded(): string | null {
+  const {blockLimitMap, blockCountMap} = Blockly;
+
+  // Ensure both maps are defined
+  if (!blockLimitMap || !blockCountMap) {
+    return null;
+  }
+
+  // Find the first instance where the limit is exceeded for a block type.
+  for (const [type, count] of blockCountMap) {
+    const limit = blockLimitMap.get(type);
+    if (limit !== undefined && count > limit) {
+      return type;
+    }
+  }
 
-export function blockLimitExceeded() {
-  return false;
+  // If no count exceeds the limit, return null.
+  return null;
 }
 
-export function getBlockLimit() {
-  return 0;
+/**
+ * Retrieves the block limit for a given block type from the block limit map.
+ * @param {string} type The type of the block to check the limit for.
+ * @returns {number} The limit for the specified block type, or 0 if not found.
```

Comment:
Do we never set the limit to 0? Is using 0 as the limit a pattern that already exists? It seems a little odd to me to mean 0 == no limit, but if we already have the logic in place we can leave it as-is. Otherwise I would do something like null or -1.

Reply:
To my surprise, we _do_ have a handful of levels where we set the limit to 0. (I would have thought in that case we'd just leave the block out of the toolbox...)

I'll try setting it up to work with `null`.